docs/user-guide/kubeconfig-file.md

```diff
@@ -94,154 +109,173 @@ users:
-      1.  current-context from the merged kubeconfig file
-      1.  Empty is allowed at this stage
-  1.  Determine the cluster info and user to use.  At this point, we may or may not have a context.  They are built based on the first hit in this chain.  (run it twice, once for user, once for cluster)
-      1.  command line argument - `user` for user name and `cluster` for cluster name
-      1.  If context is present, then use the context's value
-      1.  Empty is allowed
-  1.  Determine the actual cluster info to use.  At this point, we may or may not have a cluster info.  Build each piece of the cluster info based on the chain (first hit wins):
-      1.  command line arguments - `server`, `api-version`, `certificate-authority`, and `insecure-skip-tls-verify`
-      1.  If cluster info is present and a value for the attribute is present, use it.
-      1.  If you don't have a server location, error.
-  1.  Determine the actual user info to use. User is built using the same rules as cluster info, EXCEPT that you can only have one authentication technique per user.
-      1. Load precedence is 1) command line flag, 2) user fields from kubeconfig
-      1. The command line flags are: `client-certificate`, `client-key`, `username`, `password`, and `token`.
-      1. If there are two conflicting techniques, fail.
-  1.  For any information still missing, use default values and potentially prompt for authentication information
-
-## Manipulation of kubeconfig via `kubectl config <subcommand>`
-
-In order to more easily manipulate kubeconfig files, there are a series of subcommands to `kubectl config` to help.
-See [kubectl/kubectl_config.md](kubectl/kubectl_config.md) for help.
-
-### Example
-
-```console
-$ kubectl config set-credentials myself --username=admin --password=secret
-$ kubectl config set-cluster local-server --server=http://localhost:8080
-$ kubectl config set-context default-context --cluster=local-server --user=myself
-$ kubectl config use-context default-context
-$ kubectl config set contexts.default-context.namespace the-right-prefix
-$ kubectl config view
-```
+Before you start creating kubeconfig files, its important to understand how they are loaded and merged. The kubeconfig files in your clusters get loaded and merged based on the following:
 
-produces this output
+ * Environment variables
+ * Commands that you run from the terminal window
+ * Configuration information in each kubeconfig file
+
+The following loading and merging rules are ordered by their priority:
+
+ 1. Retrieve the kubeconfig files from disk using the following hierarchy and merge rules:
+   1. If the `CommandLineLocation` (the value of the `kubeconfig` command line option) is set, then use this file only and do not merge any other kubeconfig files. Only one instance of this flag is allowed in your kubeconfig files.
+   1. If `EnvVarLocation` (the value of $KUBECONFIG) is available, then use it to define all of the files that will be merged.
```

Comment:
Same as comment above. We should provide a pointer, more details, and maybe even an example for 'EnvVarLocation' because its not clear what that is and how to use it.